enterprise/backend/src/metabase_enterprise/serialization/v2/load.clj

```diff
@@ -102,10 +106,10 @@
                              (update :seen conj path)
                              (update :expanding disj path))
                 ;; Use the abstract path as attached by the ingestion process, not the original one we were passed.
-                rebuilt-path    (serdes/path ingested)
-                local-or-nil    (serdes/load-find-local rebuilt-path)]
+                rebuilt-path (serdes/path ingested)
+                local-or-nil (serdes/load-find-local rebuilt-path)]
             (try
-              (serdes/load-one! ingested local-or-nil)
+              (serdes/load-one! ingested (when-not require-new-entity local-or-nil))
```

Comment:
`serdes/load-find-local` requires an appdb lookup, which is wasted when `require-new-entity`. Move the `when-not` to the definition of `local-or-nil`, so we avoid doing wasted work.